e2e/notificationmenu/index.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { createNamespace, deleteNamespace } from "../utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { createWorkflow } from "~/api/tree/mutate/createWorkflow";
+import { headers } from "e2e/utils/testutils";
+import { workflowWithSecrets } from "./workflow";
+
+let namespace = "";
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("Notification Bell has an inactive state by default", async ({ page }) => {
+  await page.goto(`/${namespace}/explorer/tree`, {
+    waitUntil: "networkidle",
+  });
+
+  const notificationBell = page.getByTestId("notification-bell").nth(1);
+
+  await expect(
+    notificationBell,
+    "it renders the Notification Bell"
+  ).toBeVisible();
+
+  await notificationBell.click();
+  const notificationText = page.getByTestId("notification-text");
+
+  expect(
+    await notificationText.textContent(),
+    "the modal should display 'You do not have any notifications.'"
+  ).toMatch(/You do not have any notifications./);
+});
+
+test("Notification Bell updates depending on the count of Notification Messages", async ({
+  page,
+}) => {
+  await createWorkflow({
+    payload: workflowWithSecrets,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: "worfklow-with-secrets.yaml",
+    },
+    headers,
+  });
+
+  await page.goto(`/${namespace}/explorer/tree`, {
+    waitUntil: "networkidle",
+  });
+
+  const notificationBell = page.getByTestId("notification-bell").nth(1);
+
+  await expect(
+    notificationBell,
+    "it renders the Notification Bell"
+  ).toBeVisible();
+
+  expect(
+    page.getByTestId("notification-indicator").nth(1),
+    "the red indicator is visible"
```

Comment:
```suggestion
    "the indicator for new messages is visible"
```
Just a minor thing, but it's better to describe the semantics of the indicator. The color isn't relevant to the test (and it could change). If you change this, note that there are a multiple occurrences of this.